analysis/druidferal/src/modules/core/Snapshots2.ts

```diff
@@ -0,0 +1,166 @@
+import SPELLS from 'common/SPELLS';
+import Spell from 'common/SPELLS/Spell';
+import RACES from 'game/RACES';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Combatant from 'parser/core/Combatant';
+import Events, {
+  ApplyDebuffEvent,
+  RefreshDebuffEvent,
+  RemoveDebuffEvent,
+} from 'parser/core/Events';
+import { mergeTimePeriods } from 'parser/core/mergeTimePeriods';
+import { encodeTargetString } from 'parser/shared/modules/EnemyInstances';
+
+import { SubUptimes } from '../core/UptimeBarSubStatistic';
+
+/** Buffer in ms to use when determining if a buff was present when a DoT was applied */
+const BUFF_DROP_BUFFER = 60;
+
+export const TIGERS_FURY_SPEC: StaticSnapshotSpec = {
+  spellFunc: (_) => [SPELLS.TIGERS_FURY],
+  isActive: (_) => true,
+  isPresent: (c, timestamp) => c.hasBuff(SPELLS.TIGERS_FURY.id, timestamp, BUFF_DROP_BUFFER),
+  displayColor: '#ff9955',
+};
+
+export const PROWL_SPEC: StaticSnapshotSpec = {
+  spellFunc: (c) => {
+    const res = [];
+    res.push(SPELLS.PROWL);
+    if (c.race === RACES.NightElf) {
+      res.push(SPELLS.SHADOWMELD);
+    }
+    if (c.hasTalent(SPELLS.INCARNATION_KING_OF_THE_JUNGLE_TALENT)) {
+      res.push(SPELLS.INCARNATION_KING_OF_THE_JUNGLE_TALENT);
+    }
+    if (c.hasConduitBySpellID(SPELLS.SUDDEN_AMBUSH.id)) {
+      res.push(SPELLS.SUDDEN_AMBUSH);
+    }
+    return res;
+  },
+  isActive: (_) => true,
+  isPresent: (c, timestamp) =>
+    c.hasBuff(SPELLS.INCARNATION_KING_OF_THE_JUNGLE_TALENT.id, timestamp, BUFF_DROP_BUFFER) ||
+    c.hasBuff(SPELLS.PROWL.id, timestamp, BUFF_DROP_BUFFER) ||
+    c.hasBuff(SPELLS.PROWL_INCARNATION.id, timestamp, BUFF_DROP_BUFFER) ||
+    c.hasBuff(SPELLS.SHADOWMELD.id, timestamp, BUFF_DROP_BUFFER) ||
+    c.hasBuff(SPELLS.SUDDEN_AMBUSH_BUFF.id, timestamp, BUFF_DROP_BUFFER),
+  displayColor: '#5555ff',
+};
+
+export const BLOODTALONS_SPEC: StaticSnapshotSpec = {
+  spellFunc: (_) => [SPELLS.BLOODTALONS_TALENT],
+  isActive: (c) => c.hasTalent(SPELLS.BLOODTALONS_TALENT),
+  isPresent: (c, timestamp) => c.hasBuff(SPELLS.BLOODTALONS_BUFF.id, timestamp, BUFF_DROP_BUFFER),
+  displayColor: '#dd0022',
+};
+
+class Snapshots2 extends Analyzer {
```

Comment:
The Snapshot class needed an overhaul and some new features, but I felt it was doing too much to overhaul in one go. I plan to temporarily have this "Snapshots2" to support the new features and will over a few PRs merge stuff from the old Snapshots class into this.